tracker/settings/base.py

```diff
@@ -251,6 +254,45 @@
 # django-taggit
 TAGGIT_CASE_INSENSITIVE = True
 
+# Content Security Policy
+CSP_DEFAULT_SRC = ("'self'",)
+CSP_SCRIPT_SRC = (
+    "'self'",
+    "https://analytics.freedom.press",
+    # For Wagtail Admin
+    "'unsafe-inline'",
+    # For Twitter Widgets
+    "'unsafe-eval'",
+    "https://platform.twitter.com/widgets.js",
+    "https://platform.twitter.com/js/horizon_tweet.2bd42981e3af03ce9186a5655508da28.js",
+)
+CSP_STYLE_SRC = (
+    "'self'",
+    # For Wagtail Admin and Twitter Widgets.
+    "'unsafe-inline'",
+)
+CSP_FRAME_SRC = (
+    "'self'",
+    # For Twitter Widgets
+    "https://platform.twitter.com",
+)
+CSP_CONNECT_SRC = (
+    "'self'",
+    "https://analytics.freedom.press",
+    # For Wagtail Admin
+    "https://releases.wagtail.io/latest.txt",
+)
+CSP_IMG_SRC = (
+    "'self'",
+    "analytics.freedom.press",
+    "https://media.pressfreedomtracker.us",
+    # For Wagtail Admin
+    "http://www.gravatar.com",
+)
+
+# Report URI must be a string, not a tuple.
+CSP_REPORT_URI = os.environ.get('DJANGO_CSP_REPORT_URI',
+                                'https://freedomofpress.report-uri.com/r/d/csp/enforce')
 
```

Comment:
You will also need to set a section like this: https://github.com/freedomofpress/securedrop.org/blob/develop/securedrop/settings/base.py#L353-L361. Some of the CSP rules are set at the deployment time with environment variables. As part of that, you need to add a `CSP_OBJECT_SRC` with self for certain PDFs, etc.

Reply:
You can actually just add a section like:

```
if os.environ.get("DJANGO_CSP_MEDIA_ORIGINS"):
    csp_media_origins = os.environ["DJANGO_CSP_MEDIA_ORIGINS"].split()
    CSP_MEDIA_SRC.extend(csp_media_origins)  # video files
    CSP_IMG_SRC.extend(csp_media_origins)
    CSP_OBJECT_SRC.extend(csp_media_origins)
    CSP_CONNECT_SRC.extend(csp_media_origins)
```